File: RokidSDKDemoJava/app/src/main/java/com/rokid/mobile/sdk/demo/java/other/AsrFragment.java
package com.rokid.mobile.sdk.demo.java.other;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.rokid.mobile.lib.base.util.CollectionUtils;
import com.rokid.mobile.lib.entity.bean.remotechannel.CardMsgBean;
import com.rokid.mobile.sdk.demo.java.R;
import com.rokid.mobile.sdk.demo.java.base.BaseFragment;
import com.rokid.mobile.sdk.demo.java.base.item.CardItem;
import com.rokid.mobile.ui.recyclerview.adapter.BaseRVAdapter;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;

/**
 * Created by hongquan.zhao on 2018/3/21.
 *
 * @decription:获取card列表
 */

public class AsrFragment extends BaseFragment<AsrPresenter> {
    @BindView(R.id.fragment_device_list_btn)
    Button carListBtn;

    @BindView(R.id.fragment_device_list_rv)
    RecyclerView deviceRv;


    private BaseRVAdapter<CardItem> mAdapter;

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_device;
    }

    @Override
    protected AsrPresenter initPresenter() {
        return new AsrPresenter(this);
    }


    @Override
    protected void initVariables(View rootView, ViewGroup container, @Nullable Bundle savedInstanceState) {
        carListBtn.setText("ASR纠错");
        mAdapter = new BaseRVAdapter<>();
        deviceRv.setLayoutManager(new LinearLayoutManager(getActivity()));
        deviceRv.setAdapter(mAdapter);
    }

    @Override
    protected void initListeners() {
        carListBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                deviceRv.setVisibility(View.GONE);
                mAdapter.clearAllItemView();
                getPresenter().asrCorrectCreate("东冯破", "东风破");
            }
        });

    }

    public void showToast(String text) {
        showToastShort(text);
    }

//    public void setDeviceListData(List<CardMsgBean> itemList) {
//        if (CollectionUtils.isEmpty(itemList)) {
//            showToast(getString(R.string.fragment_card_list_empty));
//            deviceRv.setVisibility(View.GONE);
//            return;
//        }
//
//        List<CardItem> deviceItemList = new ArrayList<>();
//        for (CardMsgBean rkDevice : itemList) {
//            CardItem deviceItem = new CardItem(rkDevice);
//            deviceItemList.add(deviceItem);
//        }
//        mAdapter.setItemViewList(deviceItemList);
//        deviceRv.setVisibility(View.VISIBLE);
//    }


}

File: RokidSDKDemokotlin/app/src/main/kotlin/com/rokid/mobile/sdk/demo/other/AsrFragment.kt
package com.rokid.mobile.sdk.demo.other

import android.widget.Button
import com.rokid.mobile.lib.entity.bean.home.AsrErrorCorrectBean
import com.rokid.mobile.lib.xbase.home.callback.IGetAsrErrorCallback
import com.rokid.mobile.sdk.RokidMobileSDK
import com.rokid.mobile.sdk.demo.R
import com.rokid.mobile.sdk.demo.base.BaseFragment
import kotlinx.android.synthetic.main.device_fragment_list.view.*

/**
 * @author hongquan.zhao
 * Create Date:2018/3/21
 * Version:V1.0
 */
class AsrFragment : BaseFragment() {
    lateinit var carListBtn: Button
//    lateinit var mRecycler: RecyclerView;
//    lateinit var mAdapter: BaseRVAdapter<CardItem>;


    override fun layoutId(): Int = R.layout.device_fragment_list;


    override fun initViews() {
        carListBtn = rootView!!.fragment_device_list_btn
        carListBtn.setText(getString(R.string.fragment_asr_add_action))
//        mRecycler = rootView!!.fragment_device_list_rv

//        mRecycler.apply {
//            layoutManager = LinearLayoutManager(activity as Context?)
//            mAdapter = BaseRVAdapter()
//            adapter = mAdapter
//        }
    }

    override fun initListeners() {
        carListBtn.setOnClickListener {
            //            mRecycler.visibility = View.GONE
//            mAdapter.clearAllItemView()
            asrCorrect("东冯破", "东风破")
        }

    }

    fun asrCorrect(originText: String, newText: String) {
        RokidMobileSDK.vui.asrCorrectCreate(originText, newText, object : IGetAsrErrorCallback {
            override fun onGetAsrErrorFailed(p0: String?, p1: String?) {
                toast(getString(R.string.fragment_asr_add_fail)
                        + "\n errorCode = " + p0 + " , errorMsg =  " + p1)
            }

            override fun onGetAsrErrorSucceed(p0: AsrErrorCorrectBean?) {
                toast(getString(R.string.fragment_asr_add_success)
                        + "\n  = " + p0)
            }

        })
    }
}
